node/consensus/src/adapters/transaction_to_payload.rs

```diff
@@ -24,14 +25,18 @@ mod tests {
     use futures::sync::mpsc::channel;
     use futures::{lazy, stream};
     use primitives::signature::DEFAULT_SIGNATURE;
-    use transaction::{SendMoneyTransaction, SignedTransaction, Transaction, TransactionBody};
+    use transaction::{SendMoneyTransaction, SignedTransaction, TransactionBody};
 
     #[test]
     fn pass_through() {
         tokio::run(lazy(|| {
             let (transaction_tx, transaction_rx) = channel(1024);
             let (payload_tx, payload_rx) = channel(1024);
-            spawn_task(transaction_rx, payload_tx);
+            spawn_task(
+                transaction_rx,
+                |t| ChainPayload { transactions: vec![t], receipts: vec![] },
```

Comment:
should we pass receipts from the previous block here? or where does it get passed?

Reply:
I see passing when restarting consensus in `spawn_block_producer` but a bit unclear why it's done there instead of here.